src/components/device/ha-device-picker.ts

```diff
@@ -286,6 +285,10 @@ export class HaDevicePicker extends SubscribeMixin(LitElement) {
     ev.stopPropagation();
     const newValue = ev.detail.value;
 
+    if (newValue === "no_device") {
```

Comment:
why are you filtering out this value? Shouldn't the parent need to know that no device is picked ? Otherwise the parent data source might be out of sync with the picker.